testcases/OpTestHtxBootme.py

```diff
@@ -0,0 +1,684 @@
+    def start_htx_run(self):
+        super(HtxBootme_BlockDevice, self).start_htx_run()
+
+        path = "/usr/lpp/htx/mdt/%s" % self.mdt_file
+        if self.execute_remote_command('test -e {}'.format(path)):
+            self.fail(f"MDT file {self.mdt_file} not found")
+
+        log.debug("selecting the mdt file ")
+        cmd = f"htxcmdline -select -mdt {self.mdt_file}"
+        self.con.run_command(cmd)
+
+        if not self.all:
+            if self.is_block_device_in_mdt() is False:
+                self.fail(f"Block devices {self.block_device} are not available"
+                          f"in {self.mdt_file}")
+
+        self.suspend_all_block_device()
+
+        log.debug(f"Activating the {self.block_device}")
+        cmd = f"htxcmdline -activate {self.block_device} -mdt {self.mdt_file}"
+        self.con.run_command(cmd)
+        if not self.all:
+            if self.is_block_device_active() is False:
+                self.fail("Block devices failed to activate")
+
+        log.debug(f"Running the HTX on {self.block_device}")
+        cmd = f"htxcmdline -run -mdt {self.mdt_file}"
+        self.con.run_command(cmd)
+
+    def is_block_device_in_mdt(self):
+        """
+        verifies the presence of given block devices in selected mdt file
+        """
+        log.debug(
+            f"checking if the given block_devices are present in {self.mdt_file}")
+        cmd = f"htxcmdline -query -mdt {self.mdt_file}"
+        output = self.con.run_command(cmd)
+        device = []
+        for dev in self.block_device.split(" "):
+            if dev not in output:
+                device.append(dev)
+        if device:
+            log.debug(
+                f"block_devices {device} are not avalable in {self.mdt_file} ")
+        log.debug(
+            f"BLOCK DEVICES {self.block_device} ARE AVAILABLE {self.mdt_file}")
+        return True
+
+    def suspend_all_block_device(self):
+        """
+        Suspend the Block devices, if active.
+        """
+        log.debug("suspending block_devices if any running")
+        cmd = f"htxcmdline -suspend all  -mdt {self.mdt_file}"
+        self.con.run_command(cmd)
+
+    def is_block_device_active(self):
+        """
+        Verifies whether the block devices are active or not
+        """
+        log.debug("checking whether all block_devices are active ot not")
+        cmd = f"htxcmdline -query {self.block_device} -mdt {self.mdt_file}"
+        output = self.con.run_command(cmd)
+        device_list = self.block_device.split(" ")
+        active_devices = []
+        for line in output:
+            for dev in device_list:
+                if dev in line and "ACTIVE" in line:
+                    active_devices.append(dev)
+        non_active_device = list(set(device_list) - set(active_devices))
+        if non_active_device:
+            return False
+        log.debug(f"BLOCK DEVICES {self.block_device} ARE ACTIVE")
+        return True
+
+    def get_mpath_from_dm(self, dm_id):
```

Comment:
@abdhaleegit 3 functions  "get_mpath_from_dm" , "get_all_disk_paths" and "get_absolute_disk_path"  These 3 functions can be moved to common libraries as these are common utilities used for Block devices. These functions will be required in future for different testcases by Block devices and multipath cases on op-test.

Reply:
can be moved to OpTestUtil.py

Reply:
> can be moved to OpTestUtil.py

@abdhaleegit Tried moving the functions to common utils, but found some dependencies and code modifications, which would expand the functions a lot. Felt keeping here would be a better way to achieve this.